docs/administration-guide/gaffer-stores/federated-store.md

```diff
@@ -2,4 +2,405 @@
+
+Optionally you can add a [Graph Library](../../development-guide/project-structure/components/graph.md#graph-configuration) to store Schemas and Store Properties associated with the Federated Store sub-graphs.
+You can also add a [Cache Service](store-guide.md#caches) within the Store Properties, for example this will be required if you want to use Named Operations.
+
+### Using the REST API
+
+When using the REST API, configuration is done with the files `graphId.json` and `store.properties`. Minimal examples are shown below.
+
+```json title="graphId.json"
+{
+  "graphId": "federatedGraph",
+  "library": {
+    "class": "uk.gov.gchq.gaffer.store.library.HashMapGraphLibrary"
+  }
+}
+```
+
+```properties title="store.properties"
+gaffer.store.class=uk.gov.gchq.gaffer.federatedstore.FederatedStore
+gaffer.store.properties.class=uk.gov.gchq.gaffer.federatedstore.FederatedStoreProperties
+```
+
+### Using the Java API
+
+A `Graph` Object containing this Federated Store graph can be created as follows:
+
+```java
+final Graph federatedGraph = new Graph.Builder()
+        .config(new GraphConfig.Builder()
+                .graphId("federatedGraph")
+                .library(graphLibrary) // (1)!
+                .build())
+        .storeProperties(new FederatedStoreProperties())
+        .build();
+```
+
+1.  [See GraphLibrary Javadoc for more info](https://gchq.github.io/Gaffer/uk/gov/gchq/gaffer/store/library/GraphLibrary.html).
+
+
+### Federated Store Properties
+
+In addition to the [standard Store Properties](store-guide.md#store-properties), there are some additional properties specific to this store:
+
+- `gaffer.federatedstore.isPublicAllowed`: Controls if adding graphs with [public access](#publicprivate-access) is allowed. True by default.
+- `gaffer.federatedstore.customPropertiesAuths`: String containing auths for [allowing users to use store properties other than those contained in a graph library](#limit-custom-properties). Unset by default, allowing all users to do this.
+- `gaffer.federatedstore.storeConfiguredMergeFunctions`: Path to file containing [merge function definitions to use to override the default merge functions](#setting-default-merge-functions).
+- `gaffer.federatedstore.storeConfiguredGraphIds`: Path to file containing [Graph IDs to be queried when a user doesn't target specific graphs](#setting-default-query-graph-ids). By default, all graphs a user can access are queried.
+
+??? example "Setting these properties with the Java API"
+
+    ```java
+    final FederatedStoreProperties exampleFederatedStoreProperties = new FederatedStoreProperties();
+    exampleFederatedStoreProperties.setGraphsCanHavePublicAccess(false);
+    exampleFederatedStoreProperties.setCustomPropertyAuths("Auth1, Auth2, Auth3");
+    exampleFederatedStoreProperties.setStoreConfiguredMergeFunctions("path/to/mergefunctionDefinitions.json");
+    exampleFederatedStoreProperties.setStoreConfiguredGraphIds("path/to/defaultGraphIds.json");
+    ```
```

Comment:
Would it maybe be better to have this part split up and put under the Java API and REST API sections above?

Reply:
Could be done like that, but when using Java you will often load from a properties file anyway. The other store guides don't give Java examples at all. This one was in the old docs so I thought I'd keep it and update. It's not very important hence the default of being folded. I think it's probably more concise this way.